source/common/http/header_utility.cc

```diff
@@ -227,6 +227,29 @@ bool HeaderUtility::isEnvoyInternalRequest(const RequestHeaderMap& headers) {
          internal_request_header->value() == Headers::get().EnvoyInternalRequestValues.True;
 }
 
+void HeaderUtility::stripTrailingHostDot(RequestHeaderMap& headers) {
+  auto host = headers.getHostValue();
+  // Find last dot and return if not found or take off end if last
```

Comment:
Maybe just `// If the host ends in a period, remove it.`

Reply:
Yes fits well here, will update.